### PR TITLE
Fix Coalition Interview Mission

### PR DESCRIPTION
**Pull request by roadrunner56**
## Summary
A few months ago, I made a pull request (https://github.com/endless-sky/endless-sky/pull/6169) for a new mission involving a seminar on humanity in Coalition space. Unfortunately it was merged before Arachi had a chance to look at it. This PR aims to address issues he found with the mission. This includes fixing dialog and poorly thought out lore implications. 
## Savefile
Any file where the player has met the Coalition works. Branching dialog depends on having met the Hai, Quarg, and Korath.

### CODE REVIEW

File: data/coalition/coalition missions.txt [hunk 1]
```diff
@@ -1551,49 +1553,62 @@ mission "Saryd University Lecture"
...
 				has "First Contact: Hai: offered"
-			`	"Aliens besides the Quarg, are humans in contact with?"`
+			`	"What is the extent of contact that humans have with other aliens?"`
 			choice
-				`	"No, the only aliens humans know about are the Quarg."`
-					goto sad
+				`	"Humans have official contact with the Quarg, but the Quarg don't trade technology with humanity."`
+				`	"Humans have official contact with the Quarg, but humans and Quarg don't really intermix."`
+				`	"Humans know about the Quarg, but most people know little about them."`
+					goto quarg
```
Comment (Arachi-Lover) [inline data/coalition/coalition missions.txt:1564]: The choices at lines 1561 and 1562 should also have this goto under them, else they'll enter `label hai`

File: data/coalition/coalition missions.txt [hunk 2]
```diff
@@ -1551,49 +1553,62 @@ mission "Saryd University Lecture"
...
+				`	"Humans have official contact with the Quarg, but the Quarg don't trade technology with humanity."`
+				`	"Humans have official contact with the Quarg, but humans and Quarg don't really intermix."`
+				`	"Humans know about the Quarg, but most people know little about them."`
+					goto "quarg"
+			label quarg
+			`	The audience response is muted. You get the impression that you haven't told them anything they don't already know.`
 				goto nextquestion
 			label know
-			`	Several members of the audience are delighted. "Cooperation between different species, it is important to foster," says the Saryd who asked the question. An Arach asks you to point out where the Hai are on a galactic map, and you do so.`
+			`	Several members of the audience are delighted. "Ah, the Hai, we have heard of," says the Heliarch. "Tell us more of them, could you?"`
```
Comment (Arachi-Lover) [inline data/coalition/coalition missions.txt:1580]: Since the Hai are news to the Coalition, they'd only have heard of them if the player revealed the Hai to them in the First Contact mission. There's no condition that checks for that, though, so an `apply` `set` for something like `"coalition knows hai"` would need to be added to the option where you tell the Heliarchs about the Hai in First Contact, and that would need to be checked here with a branch.
If the player hasn't told them yet, then the condition would be applied here.

File: data/coalition/coalition missions.txt [hunk 3]
```diff
@@ -1551,49 +1553,62 @@ mission "Saryd University Lecture"
...
+			`	As you open your mouth to speak, the Heliarch cuts you off. "Enough questions on the Hai, we have." The crowd looks disappointed at this development.`
 			label nextquestion
 			branch war
 				has "FWC End: completed"
 			`	The Arach selects another, this time a Kimek. "All of human space, united politically is it?"`
 			choice
 				`	"No, human space is split between the Republic in the north and the Free Worlds in the south."`
 				`	"No, human space is split between the Republic in the west, the Free Worlds in the south, and the Syndicate in the east. The Syndicate is technically part of the Republic, but they have a lot of autonomy."`
-			`	There are murmurs in the crowd, and several begin to write down notes on various devices."`
+			`	As you answer the question, your translator begins to cut out. Repeating your answer does no good, as it contiunes to fizzle out as you talk. You look over to the Heliarch, who eyes the curious Kimek with suspicion. "Apologize for the malfunction, I must," says the Heliarch. "The next question, answer it please."`
```
Comment (Arachi-Lover) [inline data/coalition/coalition missions.txt:1589]: Why did they cut it out here? Just want to know

File: data/coalition/coalition missions.txt [hunk 4]
```diff
@@ -1551,49 +1553,62 @@ mission "Saryd University Lecture"
...
 			choice
-				`	"Yes, there are pirate planets on the fringes of human space."`
+				`	"There are entire pirate planets on the fringes of human space where warlords and criminals rule."`
 				`	"Piracy was a big problem in the past, but now it's a relatively minor issue."`
-			`	"Stamped out, may human piracy soon be," says the Arach in a solemn tone of voice. She picks a Saryd for the next question.`
+				`	"There are many independent worlds that are not ruled by one singular government. Some groups there support piracy, but others are peaceful."`
+					goto pirategood
+			`	"Stamped out, may lawlessness soon be," says the Heliarch in a solemn tone of voice. She picks a Saryd for the next question.`
+			label pirategood
+			`	Again, your microphone starts to fizzle out as you attempt to answer. The Heliarch's eight eyes are all on you, and you sense that it might not be a good idea to finish your sentence. The Heliarch selects a Saryd for the next question.`
```
Comment (Arachi-Lover) [inline data/coalition/coalition missions.txt:1611]: I might have misunderstood it, but it seems you can skip the first cutoff? If so, the `Again` would be odd if you did skip it.

File: data/coalition/coalition missions.txt [hunk 5]
```diff
@@ -1617,8 +1632,8 @@ mission "Saryd University Lecture"
...
-			`	The crowd shows considerable interest at this last part. "Knowledge of the jump drive, a species besides the Quarg possesses? Great news, this is! Our struggle against the Quarg, we do not take up alone. Escape the cages of the Drak, do any other species?" The Arach does not seem to mind you answering another question.`
+			`	"I've met an alien species there called the Korath. They're reptillian creatures. What's left of their planets has been ruined by infighting. The Kor Sestor and Kor Mereti have been fighting a civil war using autonomous ships. The Kor Efreti have stayed out of the civil war, living under the Quarg's protection. The Korath Exiles, have been banished to the Core by the Quarg, for refusing to give up their ability to manufacture jump drives. They are now raiding humanity for supplies."`
```
Comment (Arachi-Lover) [inline data/coalition/coalition missions.txt:1635]: `The Korath Exiles, have been banished to the Core by the Quarg` they weren't banished by the Quarg, it was the Drak/Archons that gave the Korath their ultimatum, and enforced the Exile.

File: data/coalition/coalition missions.txt [hunk 6]
```diff
@@ -1617,8 +1632,8 @@ mission "Saryd University Lecture"
...
+			`	The crowd shows considerable interest at this last part. "Knowledge of the jump drive, a species besides the Quarg possesses? Great news, this is! Our struggle against the Quarg, we do not take up alone. Possess jump drives, do any other species?" The Heliarch does not seem to mind you answering another question.`
 			branch unfettered
```
Comment (Arachi-Lover) [inline data/coalition/coalition missions.txt:1637]: This part would need the player to explain to the Coalition who the Hai are, in case they haven't yet, so would need that check for a condition again (to see if the player has revealed the Hai to the Coalition). If you do reveal them here it'd need the condition applied too, though you could just check for the condition and not give the player the chance to talk about the Unfettered if they haven't revealed the Hai, to sort of continue with the choice to not reveal them.

**Review by Arachi-Lover — COMMENTED** [review]
Thanks for the fixes!
Had a few suggestions on some of them.

File: data/coalition/coalition missions.txt [hunk 7]
```diff
@@ -1487,19 +1487,19 @@ mission "Saryd University Lecture"
...
 			label accept
-			`	"Wonderful!" he says. "Aelbris, I am." He soon gathers his things and boards your ship. As he enters, he seems fascinated by its structure and design. His gawking makes it difficult to get him to finally settle down into his bunk.`
+			`	"Thank you! Aelbris, my name is. To be late I do not wish, as not recorded, the event is." He soon gathers his things and boards your ship.`
 					accept
 	on complete
 		"coalition jobs" += 3
 		payment 25000
 		conversation
-			`During the trip, Aelbris was polite, if a little excited to have a human as his pilot. He asked many questions during the flight, which you answered as best you could. Once you and your passenger depart from your ship, Aelbris turns around.`
+			`	Once you and your passenger depart from your ship, he turns around.`
```
[the remark this answers is not in the document]
Comment (roadrunner56) [reply data/coalition/coalition missions.txt:1502]: > During the trip, Aelbris was polite, if a bit intrusive in his questions about your homeworld. You answered as best you could, though at times you struggled to understand what was even being asked. Once you ...

That line sounds really good actually, would you mind if I copy and paste that into the mission?

[the remark this answers is not in the document]
Comment (Arachi-Lover) [reply data/coalition/coalition missions.txt:1502]: Sure, if you're ok with that it's fine.

File: data/coalition/coalition missions.txt (hunk 3 above)
Comment (roadrunner56) [reply data/coalition/coalition missions.txt:1589]: My reasoning was that the Heliarchs may not want civilians to know that humans can function fine without a united government. I don't know if that's a misinterpretation of what Heliarchs would be concerned about, though.

Comment (Arachi-Lover) [reply data/coalition/coalition missions.txt:1589]: It's fine, I just wasn't sure on the reason.

File: data/coalition/coalition missions.txt (hunk 5 above)
Comment (roadrunner56) [reply data/coalition/coalition missions.txt]: Sorry, completely missed that.

File: data/coalition/coalition missions.txt (hunk 2 above)
Comment (roadrunner56) [reply data/coalition/coalition missions.txt]: I tried adding a branch depending on whether or not the Coalition know about the Hai. However, I'm very new to using `apply` `set`, so I might have messed something up.

Comment (Arachi-Lover) [reply data/coalition/coalition missions.txt]: Looks good, however I don't know if `action` works? I thought it was only `apply` that did that, always saw that one be used.

Comment (roadrunner56) [reply data/coalition/coalition missions.txt]: I just used what I saw in the wiki: https://github.com/endless-sky/endless-sky/wiki/WritingConversations#action
I think it became a thing in 0.9.15.

Comment (Arachi-Lover) [reply data/coalition/coalition missions.txt]: `Prior to v. 0.9.15 the action node was named apply`
Ah, I see.

In that case just make it so that you also get that condition if you reveal the Hai during the seminar, and should be good.

Comment (roadrunner56) [reply data/coalition/coalition missions.txt]: I think I did it now.

**Review by Arachi-Lover — APPROVED** [review]

**Comment by roadrunner56** [comment]
One more thing - I mention the Drak in the Korath section, but that never really leads anywhere. It feels awkward to mention this entirely new alien species and then no one asks about it or even acknowledges it. I'm not really sure how to solve this, though. Should I cut out the reference entirely?

**Comment by Arachi-Lover** [comment]
It's fine there as the it's player mentions it, my gripe with what you had was that the Coalition were mentioning them so nonchalantly, and had that Unfettered view on them like I said in my review.
The Coalition knows of the Drak/Archons, so it's not like you're revealing yet another group to them.
If you want to rephrase it to remove the mention you could (though it might get a bit awkward to explain the Korath situation without explicit mention of them), but I don't think it's needed.

File: data/coalition/coalition missions.txt [hunk 8]
```diff
@@ -1551,49 +1555,70 @@ mission "Saryd University Lecture"
...
+				`	"Humans know about the Quarg, but most people know little about them."`
+			label quarg
+			`	The audience response is muted. You get the impression that you haven't told them anything they don't already know.`
 				goto nextquestion
 			label know
-			`	Several members of the audience are delighted. "Cooperation between different species, it is important to foster," says the Saryd who asked the question. An Arach asks you to point out where the Hai are on a galactic map, and you do so.`
+			branch coalitionknow
+				has "coalition knows hai"
+			`	"Several members of the audience are delighted. "Unknown to us, this species is," says the Heliarch. "That you have made contact with them despite the Quarg, good news it is."`
+			`	"Tell us more of them, could you?" asks another scientist. As you open your mouth to speak, the Heliarch cuts you off. "Enough questions on the Hai, we have." The crowd looks disappointed at this development.`
```
Comment (W1zrad) [inline data/coalition/coalition missions.txt:1586]:
```suggestion
			`	"Tell us more of them, could you?" asks another scientist.`
			`	As you open your mouth to speak, the Heliarch cuts you off. "Enough questions on the Hai, we have." The crowd looks disappointed at this development.`
```
Can't have dialogue from two different speakers in the same paragraph.

File: data/coalition/coalition missions.txt [hunk 9]
```diff
@@ -1551,49 +1555,70 @@ mission "Saryd University Lecture"
...
+			action
+				set "coalition knows hai"
+			label coalitionknow
+			`	Several members of the audience are delighted. "Ah, the Hai, we have heard of," says the Heliarch. "That you have made contact with them despite the Quarg, good news it is."`
+			`	"Tell us more of them, could you?" asks another scientist. As you open your mouth to speak, the Heliarch cuts you off. "Enough questions on the Hai, we have." The crowd looks disappointed at this development.`
```
Comment (W1zrad) [inline data/coalition/coalition missions.txt:1591]:
```suggestion
			`	"Tell us more of them, could you?" asks another scientist.`
			`	As you open your mouth to speak, the Heliarch cuts you off. "Enough questions on the Hai, we have." The crowd looks disappointed at this development.`
```

File: data/coalition/coalition missions.txt [hunk 10]
```diff
@@ -1617,11 +1642,13 @@ mission "Saryd University Lecture"
...
-			`	The crowd shows considerable interest at this last part. "Knowledge of the jump drive, a species besides the Quarg possesses? Great news, this is! Our struggle against the Quarg, we do not take up alone. Escape the cages of the Drak, do any other species?" The Arach does not seem to mind you answering another question.`
+			`	"I've met an alien species there called the Korath. They're reptillian creatures. What's left of their planets has been ruined by infighting. The Kor Sestor and Kor Mereti have been fighting a civil war using autonomous ships. The Kor Efreti have stayed out of the civil war, living under the Quarg's protection. The Korath Exiles have been banished to the Core by the Drak, for refusing to give up their ability to manufacture jump drives. They are now raiding humanity for supplies."`
```
Comment (W1zrad) [inline data/coalition/coalition missions.txt:1645]:
```suggestion
			`	"I've met an alien species there called the Korath," you say. "They're reptillian creatures. What's left of their planets has been ruined by infighting. The Kor Sestor and Kor Mereti have been fighting a civil war using autonomous ships. The Kor Efreti have stayed out of the civil war, living under the Quarg's protection. The Korath Exiles have been banished to the Core by the Drak for refusing to give up their ability to manufacture jump drives. They are now raiding humanity for supplies."`
```
Just a quick grammar fix, and added a dialogue tag to make it clear this is `<first>` speaking.

**Review by W1zrad — CHANGES_REQUESTED** [review]
Just a few small fixes. I really like the bit about your translator "malfunctioning" to keep specific information from getting out - clever!

**Review by W1zrad — APPROVED** [review]
Looks solid to me!

**Comment by roadrunner56** [comment]
> I really like the bit about your translator "malfunctioning" to keep specific information from getting out - clever!
Ha, thank Arachi for that one, he came up with it.
